server/src/main/java/org/apache/druid/indexing/overlord/DataSourceMetadata.java

```diff
@@ -69,6 +69,15 @@ public interface DataSourceMetadata
    */
   boolean matches(DataSourceMetadata other);
 
+  /**
+   * Returns true if the metadata in this instance is greater than the metadata in "other"
+   *
+   * Behavior is undefined if you pass in an instance of a different class from this one.
```

Comment:
this should be documented in the interface rather than the implementing class itself, since the caller wouldn't necessarily be aware of it otherwise.